### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by dodoburner**
In this PR I have:

- Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.
- Created the following methods:
1. Users
A method that returns the 3 most recent posts for a given user.
2. Posts
A method that updates the posts counter for a user.
A method which returns the 5 most recent comments for a given post.
3. Comments
A method that updates the comments counter for a post.
4. Likes
A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  has_many :likes
+  has_many :comments
+
+  # def initialize
+  #   update_posts_counter
+  # end
+
+  def update_posts_counter
+    User.find(author_id).update(posts_counter: Post.where(author_id: author.id).count)
+  end
```
Comment (OLIPLICHE) [inline app/models/post.rb]: - [ ] **Optional suggestion**: 
- You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to a database, then make the `update_post_counter` method private:
- 
//  after_save :update_post_counter

    // private

// def update_post_counter
    author.increment!(:posts_counter)
end

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comments_counter
+    Post.find(post_id).update(comments_counter: Comment.where(post_id: post.id).count)
+  end
```
Comment (OLIPLICHE) [inline app/models/comment.rb]: - [ ]  Optional Suggestion: Just like what I suggested `after_save :update_comments_counter`

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_likes_counter
+    Post.find(post_id).update(likes_counter: Like.where(post_id: post.id).count)
+  end
```
Comment (OLIPLICHE) [inline app/models/like.rb]: - [ ]  Optional Suggestion: Just like what I suggested `after_save :update_likes_counter`

**Review by OLIPLICHE — APPROVED** [review]
Hi @dodoburner, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Creates and sets up associations between models (✅)) 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (dodoburner) [reply app/models/post.rb]: Thank you very much for your suggestion, I was trying to do that by creating an initialize method and calling it from there but it wasn't working. @OLIPLICHE

Comment (OLIPLICHE) [reply app/models/post.rb]: Okay, Move on, don't waste time. you can try again in the milestone ahead

**Pull request merged**